Initial Refactor

dirtycajunrice opened the pull request:
Overview:
* restructure repository to plugin as root
* Build.cs: remove redundant definitions, simplify library import, format
* uplugin: remove engine version in source code
* module.h/cpp: remove empty .h file definition and set cpp file as defaultmoduleimpl
* converted from actor to handle + subsystem + function library
* general cleanup of deprecated functions
* Helpers added in many places to simplify execution flow
* async task created for oauth login
* many inputs migrated to native

0xFirekeeper commented:
Hey @dirtycajunrice - first of all I want to say this is a fantastic refactor, highly appreciated!

To get this to the finish line I added a few things I noticed in here while testing.

Furthermore, in `BP_ThirdwebActor` wise I think we're not reusing this email variable, probably should be like this
![3f0d55ebaf607400bf067262938bbe24](https://github.com/user-attachments/assets/58ff97df-9ffa-445d-8b69-ad14aac9cb0d)

The Level_Thirdweb_OAuth also appears to always throw invalid wallet type for me, were you able to run the levels successfully?

Thanks again!

0xFirekeeper commented on Source/Thirdweb/Private/ThirdwebSubsystem.cpp:
```diff
@@ -0,0 +1,178 @@
+}
+
+
+EFunctionResult UThirdwebSubsystem::BP_CreateSmartWallet(const FWalletHandle& LocalWallet, FWalletHandle& SmartWallet, FString& Error, const int64 ChainID, const bool bGasless, const FString& Factory,
```
Can we call this PersonalWallet - to avoid confusion with our previous product LocalWallet which is equivalent to PrivateKeyWallet. A Smart Wallet can take in any type of wallet (handle) and work.

dirtycajunrice commented:
Addressed all review points, and added beginnings of widget example

https://github.com/user-attachments/assets/d63c57c5-74cd-4da9-b88a-29395524d8fc
